plugins/ingest/src/main/java/org/elasticsearch/ingest/Data.java

```diff
@@ -129,4 +129,33 @@ public String getId() {
     public boolean isModified() {
         return modified;
     }
+
+    public Map<String, Object> asMap() {
```

Comment:
again, this is not possible, sorry for the confusion. I would go for an enhanced Data as part of the plugin that implements ToXContent.